pkg/policy/policies_compute.go

```diff
@@ -126,9 +126,20 @@ func (ps *policies) updateUserlandPolicies() {
 			continue
 		}
 
-		if p.DataFilter.Enabled() ||
-			p.RetFilter.Enabled() ||
-			p.ScopeFilter.Enabled() ||
+		hasUserlandFilters := false
+
+		// Check filters under Rules
+		for _, rule := range p.Rules {
+			if rule.DataFilter.Enabled() ||
+				rule.RetFilter.Enabled() ||
+				rule.ScopeFilter.Enabled() {
+				hasUserlandFilters = true
+				break
+			}
+		}
```

Comment:
Could this logic set an internal flag to be checked via a Rules public method as a shortcut in the pipeline?

Reply:
We already set `ps.filterableInUserland` a few lines below and use it in the pipeline by `FilterableInUserland()`, what else do you suggest?

Reply:
We already have a short circuit:

https://github.com/aquasecurity/tracee/blob/8d22a0791f98d4a1df4e30c95bdb9d9eb3c6801f/pkg/ebpf/events_pipeline.go#L307-L311

But I think it would be faster if we only check a boolean (the `hasUserlandFilters` promoted as a Policies field). So instead of acquiring mutex and doing the bitwise, we would only be required to acquire and return true or false.

https://github.com/aquasecurity/tracee/blob/8d22a0791f98d4a1df4e30c95bdb9d9eb3c6801f/pkg/policy/policy_manager.go#L565-L570

Also, as `PolicyManager` has its own mutex, these atomic aren't necessary anymore (reminiscences of another era):

https://github.com/aquasecurity/tracee/blob/8d22a0791f98d4a1df4e30c95bdb9d9eb3c6801f/pkg/policy/policies.go#L88-L90

Reply:
But this `hasUserlandFilters` variable doesn't takes the UID and PID filters into account as done below, so what do you suggest exactly?
Anyway, I'm not saying we shouldn't do such a shortcut to avoid mutexes, just that I'm not sure it is related to this PR that handles policy structure change

Reply:
We already discussed this offline. 👍🏼